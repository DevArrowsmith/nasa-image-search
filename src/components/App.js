import React, { useState } from 'react';
import { GlobalStyles } from '../styles/global';
import Header from './Header';
import LoaderSpinner from './LoaderSpinner';
import SearchResults from './SearchResults';

function App() {

  const [searchResults, setSearchResults] = useState([]);
  const [resultsLoading, setResultsLoading] = useState(false);

  console.log(searchResults);

  return (
    <>
      <GlobalStyles />

      <Header
        setSearchResults = {setSearchResults}
        setResultsLoading = {setResultsLoading}
       />

      {resultsLoading && 
        <LoaderSpinner />
      }

      {!resultsLoading &&
       <SearchResults
          searchResults = {searchResults} 
        />
      }

    </>
  )

}

export default App;
