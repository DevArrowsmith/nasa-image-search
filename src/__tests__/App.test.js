import React from 'react';
import { render, getByTestId } from '@testing-library/react';
import App from '../components/App';

test('component App renders', () => {
  const { asFragment } = render(<App />);
  const component = asFragment();
  expect (component).toMatchSnapshot();
});

test('component App renders a loading spinner when resultsLoading is true', () => {
  const { getByTestId } = render( <App /> );
  expect(getByTestId("Loader")).toBeInTheDocument();
})
