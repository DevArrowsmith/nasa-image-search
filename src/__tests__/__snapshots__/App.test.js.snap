// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component App renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhUy cJaanl"
  >
    <img
      alt="NASA logo"
      class="sc-AxjAm ivsrOq"
      src="./images/nasa.png"
    />
    <div
      class="sc-AxirZ kiBIEj"
    >
      <input
        class="sc-AxiKw iuhRGG"
        placeholder="Search"
        type="search"
      />
      <img
        alt="Click here to search"
        class="sc-AxhCb fasBgv"
        src="./images/search.png"
      />
    </div>
  </div>
  <div
    class="sc-AxgMl cylSDq"
  />
</DocumentFragment>
`;
